core/src/main/java/eu/fasten/core/data/callableindex/ExtendedGidGraph.java

```diff
@@ -75,14 +75,25 @@ public ExtendedGidGraph(long index, String product, String version, List<Long> n
     public JSONObject toJSON() {
         var json = super.toJSON();
         var callSitesInfo = new JSONObject();
-        getCallsInfo().forEach((edge, info) -> {
-            var edgeStr = String.format("[%d, %d]", edge.getFirst(), edge.getSecond());
-            var infoJson = new JSONObject();
-            infoJson.put("line", info.getLine());
-            infoJson.put("receiver_type_ids", new JSONArray(Arrays.asList(info.getReceiverTypeIds())));
-            infoJson.put("call_type", info.getCallType().getLiteral());
-            callSitesInfo.put(edgeStr, infoJson);
-        });
+        try {
```

Comment:
Thanks for looking into a fix. I'm not too happy with the chosen way to fix it, however. These null pointer exceptions will happen for each C and Python graph. By catching and then fixing the issue, I feel we are dealing with this too late. Exceptions should really be reserved for unexpected situations. Can we fix this earlier in the code?

Reply:
Thanks @MagielBruntink  for the immediate review, I changed the approach. More specifically I created a separated toJson() method for C & Python packages, which is the exact same with the toJson() method but instead of acessing null values in Call sites records, it just inserts null values to the callsites_info Json attributes. This method is employed when thei C & Python metadata Plugins are employed. Now the Exception will not be triggered